### PR TITLE
[SYCL] Remove old style interop

### PR DESCRIPTION
**Pull request by alexbatashev**

### CODE REVIEW

File: SYCL/Basic/queue/queue.cpp
Comment (s-kanaev) [inline SYCL/Basic/queue/queue.cpp:26]: The test should include `<CL/sycl/backend/opencl.hpp>` now.
This is relevant for other modified tests also.
I'm not sure about this, but, probably, the test should be linked against OpenCL library also.

**Review by s-kanaev — CHANGES_REQUESTED** [review]

File: SYCL/OnlineCompiler/online_compiler_OpenCL.cpp
Comment (alexbatashev) [inline SYCL/OnlineCompiler/online_compiler_OpenCL.cpp]:
```suggestion
```

**Review by v-klochkov — DISMISSED** [review]

File: SYCL/DeprecatedFeatures/buffer_interop.cpp
Comment (mkinsner) [inline SYCL/DeprecatedFeatures/buffer_interop.cpp:3]: Which APIs necessitate this macro in the tests?

Comment (alexbatashev) [reply SYCL/DeprecatedFeatures/buffer_interop.cpp:3]: We remove some APIs by wrapping them with `#ifdef __SYCL_INTERNAL_API`, in this particular case old OpenCL-interop APIs are removed (since tests and impls are in different repos, we have to commit test changes first).

**Review by mkinsner — DISMISSED** [review]

File: SYCL/Basic/fpga_tests/fpga_queue.cpp
Comment (gmlueck) [inline SYCL/Basic/fpga_tests/fpga_queue.cpp:14]: According to the SYCL 2020 spec, the OpenCL interop header is named `sycl/backend/opencl.hpp` (without the `CL`).  Do we support that naming?

Comment (alexbatashev) [reply SYCL/Basic/fpga_tests/fpga_queue.cpp:14]: There’s a separate pull request for that change.

Comment (gmlueck) [reply SYCL/Basic/fpga_tests/fpga_queue.cpp:14]: OK

**Review by gmlueck — DISMISSED** [review]

**Review by v-klochkov — DISMISSED** [review]

File: SYCL/DeprecatedFeatures/queue_old_interop.cpp
Comment (vladimirlaz) [inline SYCL/DeprecatedFeatures/queue_old_interop.cpp]: wrong naming

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Comment by vladimirlaz** [comment]
Submitted as the only comment has changed after all approvals had been received.

**Pull request merged**
